sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4552,7 +4542,9 @@ object SQLConf {
       RemovedConfig("spark.sql.ansi.strictIndexOperator", "3.4.0", "true",
         "This was an internal configuration. It is not needed anymore since Spark SQL always " +
           "returns null when getting a map value with a non-existing key. See SPARK-40066 " +
-          "for more details.")
+          "for more details."),
+      RemovedConfig("spark.sql.hive.verifyPartitionPath", "4.0.0", "false",
+        s"This config was replaced by '${SPARK_IGNORE_MISSING_FILES.key}'.")
```

Comment:
shall we use the sql config since https://github.com/apache/spark/pull/43097 will be merged soon?

Reply:
Yep, I plan to merge this PR https://github.com/apache/spark/pull/43097 first of all, and then rebase this PR on those changes by modifying the test and this place.

Reply:
rebased and changed the config name.